MidCamp/midcamp
# Multi Year Support

# Description

This PR adds multi-year support to MidCamp.  It includes:

- Adds `field_event` to all content types so they can be associated with a yearly Midcamp term
- Adds a node's Event URL path to the pathauto pattern, resulting in node aliases like `/2018/article/article-title`
- Adds an events filter and column to the admin content view to administer content by event
- Custom module `midcamp_event_label` to expose the `field_event` value from a node in a block (used to place above the title, which is not currently available as a field in entity display settings...see [D.O. Issue 2353867](https://www.drupal.org/project/drupal/issues/2353867)).  Creates a sort of "breadcrumb" to let you know what year's event the content is associated with.
- Adds a menu called "Midcamp 2019 Navigation" so each event year can have discrete navigation.
    - Configures this new navigation to appear on all pages except 2018 year content pages
- Updates "Main Menu" block to only show on pages starting with the path `/2018`

# To test

- Import config with `drush cim -y`
- Ensure a "Midcamp 2019" term is created at http://midcamp.org.docker.amazee.io/admin/structure/taxonomy/manage/event/overview.  Give it an alias of `/2019`.
- Ensure that the existing "Midcamp 2018" term has an alias of `/2018`
- Bulk regenerate all content aliases at http://midcamp.org.docker.amazee.io/admin/config/search/path/update_bulk
    - Observe that articles now include the Midcamp 2018 taxonomy term path as the first component of their path alias.
- Populate the "Midcamp 2019 Navigation" menu with some dummy links
- Create some new pages of various content types.  Observe that when you tag them in `field_event` to Midcamp 2018 or Midcamp 2019, you will see:
    - The URL alias incorporate the event term's path pattern
    - The "MidCamp Event Label Block" positioned above the node title, reflecting the correct event assignment
    - Observe that `2018` content shows the "Main Menu", while `2019` content shows the "Midcamp 2019 Navigation" menu

# Screenshots

![image](https://user-images.githubusercontent.com/4048700/46185942-bfe85e00-c2a1-11e8-95fd-e668e3e19330.png)
Some 2018 content.

![image](https://user-images.githubusercontent.com/4048700/46186028-279ea900-c2a2-11e8-9631-e22b1095a7ec.png)
Example 2019 content.

# Next Steps

In the production environment, we will need to:

- Deploy this code
- Create the event term for Midcamp 2019
- Regenerate all content path aliases
- Populate the new "Midcamp 2019 Navigation" menu
- Write some content
